### PR TITLE
sent message on supplier updates

### PR DESCRIPTION
**Pull request by DFSko**
happy-travel/agent-app-project#1501

### CODE REVIEW

**Review by AlexeyShok — APPROVED** [review]

File: HappyTravel.Sunpu.Api/Infrastructure/Constants/MessageBusTopics.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace HappyTravel.Sunpu.Api.Infrastructure.Constants;
+
+public static class MessageBusTopics
```
Comment (suhan3z) [inline HappyTravel.Sunpu.Api/Infrastructure/Constants/MessageBusTopics.cs:3]: I think we should also expose the format of the messages in the message bus for swagger

Comment (DFSko) [reply HappyTravel.Sunpu.Api/Infrastructure/Constants/MessageBusTopics.cs:3]: Added to interface

File: HappyTravel.Sunpu.Api/Infrastructure/ModelExtensions/RichSupplierExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+using HappyTravel.Sunpu.Api.Models;
+
+namespace HappyTravel.Sunpu.Api.Infrastructure.ModelExtensions;
+
+public static class RichSupplierExtensions
+{
+    public static SlimSupplier ToSlimSupplier(this RichSupplier richSupplier)
```
Comment (suhan3z) [inline HappyTravel.Sunpu.Api/Infrastructure/ModelExtensions/RichSupplierExtensions.cs:7]: Lack of comment here

File: HappyTravel.Sunpu.Api/Services/MessageBus.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+public class MessageBus : IMessageBus
+{
+    public MessageBus(IConnection connection)
+    {
+        _connection = connection;
+    }
+
+
+    public void Publish<T>(string topicName, T message)
```
Comment (suhan3z) [inline HappyTravel.Sunpu.Api/Services/MessageBus.cs:14]: And here

File: HappyTravel.Sunpu.Api/Services/SupplierService.cs [hunk 4]
```diff
@@ -149,11 +153,14 @@ static Result Validate(RichSupplier richSupplier)
     }
 
 
-    public Task<Result> Delete(string supplierCode, CancellationToken cancellationToken)
+    public async Task<Result> Delete(string supplierCode, CancellationToken cancellationToken)
```
Comment (suhan3z) [inline HappyTravel.Sunpu.Api/Services/SupplierService.cs:156]: And here

File: HappyTravel.Sunpu.Api/Services/SupplierService.cs [hunk 5]
```diff
@@ -170,14 +177,17 @@ Task RefreshStorage()
     }
 
 
-    public Task<Result> SetEnableState(string supplierCode, EnableState enableState, string reason, CancellationToken cancellationToken)
+    public async Task<Result> SetEnableState(string supplierCode, EnableState enableState, string reason, CancellationToken cancellationToken)
```
Comment (suhan3z) [inline HappyTravel.Sunpu.Api/Services/SupplierService.cs:180]: And here

**Pull request merged**